pipelines/zendesk/helpers/api_helpers.py

```diff
@@ -0,0 +1,99 @@
+    @:param disable_cache: bool, If true disables the caching of already retrieved objects.
+    @:param domain: str, domain of your Zendesk company account
+    @:param proactive_ratelimit: int, user imposed rate limit due to budgeting issues, etc.
+    @:param proactive_ratelimit_request_interval: int, indicates how many seconds to wait when rate limit is reached, default is 10 seconds
+    @:param ratelimit_budget: int that sets the maximum amount of time that the user can spend being rate limited
+    @:param timeout: float that sets the global timeout on Zenpy
+    @:returns: API client to make requests to Zendesk API
+    """
+    # oauth token is the preferred way to authenticate, followed by api token and then email + password combo
+    # if none of these are filled, simply return an error
+    if isinstance(credentials, ZendeskCredentialsOAuth):
+        zendesk_client = Zenpy(
+            subdomain=credentials.subdomain, oauth_token=credentials.oauth_token,
+            domain=domain, timeout=timeout, ratelimit_budget=ratelimit_budget, proactive_ratelimit=proactive_ratelimit, proactive_ratelimit_request_interval=proactive_ratelimit_request_interval,
+            disable_cache=disable_cache
+        )
+    elif isinstance(credentials, ZendeskCredentialsToken):
+        zendesk_client = Zenpy(
+            token=credentials.token, email=credentials.email, subdomain=credentials.subdomain,
+            domain=domain, timeout=timeout, ratelimit_budget=ratelimit_budget, proactive_ratelimit=proactive_ratelimit, proactive_ratelimit_request_interval=proactive_ratelimit_request_interval,
+            disable_cache=disable_cache
+        )
+    elif isinstance(credentials, ZendeskCredentialsEmailPass):
+        zendesk_client = Zenpy(
+            email=credentials.email, subdomain=credentials.subdomain, password=credentials.password,
+            domain=domain, timeout=timeout, ratelimit_budget=ratelimit_budget, proactive_ratelimit=proactive_ratelimit, proactive_ratelimit_request_interval=proactive_ratelimit_request_interval,
+            disable_cache=disable_cache
+        )
+    return zendesk_client
+
+
+def process_ticket(ticket: Ticket, custom_fields: DictStrStr, pivot_fields: bool = True) -> DictStrAny:
+    """
+    Helper that returns a dictionary of the ticket class provided as a parameter. This is done since to_dict()
+    method of Ticket class doesn't return all the required information
+    @:param sample_ticket: Ticket Object returned by a Zenpy API call, contains info on a single ticket.
+    @:param custom_fields: A dict containing all the custom fields available for tickets.
+    @:param pivot_fields: Bool that indicates whether to pivot all custom fields or not.
+    @:return base_dict: A dict containing 'cleaned' data about a ticket.
+    """
+
+    base_dict: DictStrAny = ticket.to_dict()
+    # get tags as a string eliminating the square brackets from string
+    base_dict["tags"] = str(base_dict["tags"])[1:-1]
```

Comment:
was this in the requirements? what is the problem with a list? if you want to generate a JSON type column that will be stored as struct in the destination add `tags` to `columns` in the resource declaration with type "complex" ie.
`columns={"tags": {"data_type": "complex"}}`

if that was your intention then tell me if the above does not work. I'd like to try it out :)